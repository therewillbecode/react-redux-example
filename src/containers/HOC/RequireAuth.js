import React, { Component } from "react";
import { connect } from "react-redux";

export class AuthenticatedComponent extends Component {
  componentWillMount() {
    if (!this.props.authenticated) {
      this.props.history.replace("/");
    }
  }

  componentWillUpdate(nextProps) {
    if (!nextProps.authenticated) {
      this.props.history.replace("/");
    }
  }

  render() {
    const ComposedComponent = this.props.composedComponent;
    return (
      <div className="authenticated">
        {this.props.authenticated ? (
          <ComposedComponent {...this.props} />
        ) : null}
      </div>
    );
  }
}

function RequireAuth(ComposedComponent) {
  const mapStateToProps = state => ({
    authenticated: state.auth.get("isAuthenticated"),
    composedComponent: ComposedComponent
  });

  return connect(mapStateToProps)(AuthenticatedComponent);
}

export default RequireAuth;
