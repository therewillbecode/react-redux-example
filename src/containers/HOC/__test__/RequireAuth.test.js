/* eslint-disable */

import React from "react";
import { shallow, mount } from "enzyme";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import createHistory from "history/createBrowserHistory";
import { fromJS } from "immutable";

import RequireAuth, { AuthenticatedComponent } from "../RequireAuth";

const Component = () => <div />;
Component.displayName = "CustomComponent";
const mockStore = configureStore([]);

describe("RequireAuth", () => {
  const RequireAuthComponent = RequireAuth(Component);
  const initialState = { auth: fromJS({ isAuthenticated: true }) };
  let wrapper = null;

  beforeEach(() => {
    wrapper = mount(
      <Provider store={mockStore(initialState)}>
        <RequireAuthComponent />
      </Provider>
    );
  });

  it("should return a component", () => {
    expect(wrapper.find("AuthenticatedComponent")).toHaveLength(1);
  });

  it("should pass correct props", () => {
    const tree = renderer.create(<wrapper />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("rendering", () => {
  let history = null;

  describe("is authenticated", () => {
    history = createHistory();
    const wrapper = shallow(
      <AuthenticatedComponent
        composedComponent={Component}
        authenticated
        history={history}
      />
    );

    it("should render the passed component", () => {
      expect(wrapper.find("CustomComponent")).toHaveLength(1);
    });
  });

  describe("is not authenticated", () => {
    const wrapper = shallow(
      <AuthenticatedComponent
        composedComponent={Component}
        authenticated={false}
        history={history}
      />
    );

    it("should not render the passed component", () => {
      expect(wrapper.find("CustomComponent")).toHaveLength(0);
    });
  });
});
