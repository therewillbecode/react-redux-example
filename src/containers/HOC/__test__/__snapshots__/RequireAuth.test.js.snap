// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HOC should pass correct props 1`] = `<wrapper />`;

exports[`RequireAuth should pass correct props 1`] = `<wrapper />`;
